scheduler/pkg/agent/server.go

```diff
@@ -419,9 +419,6 @@ func (s *Server) StopAgentStreams() {
 }
 
 func (s *Server) syncMessage(request *pb.AgentSubscribeRequest, stream pb.AgentService_SubscribeServer) error {
-	s.mutex.Lock()
```

Comment:
Not sure why we are removing this lock

Reply:
`syncMessage` is calling 
- `AddServerReplica` that has already a lock internally (but outside of  `PublishModelEvent`)
- `ScheduleFailedModels` that looks not required a lock

Mainly removing the lock so that `PublishModelEvent` is called with no locks held but also it doesnt look that `syncMessage` requires a lock as explained above.